samtranslator/model/eventsources/pull.py

```diff
@@ -287,7 +287,26 @@ def get_policy_arn(self):  # type: ignore[no-untyped-def]
         return ArnGenerator.generate_aws_managed_policy_arn("service-role/AWSLambdaMSKExecutionRole")  # type: ignore[no-untyped-call]
 
     def get_policy_statements(self):  # type: ignore[no-untyped-def]
-        return None
+        document = None
+        if self.SourceAccessConfigurations:
+            for conf in self.SourceAccessConfigurations:
+                if isinstance(conf, dict) and conf.get("Type") == "CLIENT_CERTIFICATE_TLS_AUTH" and conf.get("URI"):
```

Comment:
So it's quickly clear why this specific policy:

```suggestion
                # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-lambda-eventsourcemapping-sourceaccessconfiguration.html
                if isinstance(conf, dict) and conf.get("Type") == "CLIENT_CERTIFICATE_TLS_AUTH" and conf.get("URI"):
```